services/wallet/thirdparty/coingecko/client_test.go

```diff
@@ -0,0 +1,98 @@
+package coingecko
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func setupTest(t *testing.T, delay time.Duration, response []byte) (*httptest.Server, func()) {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		time.Sleep(delay * time.Second)
+		w.WriteHeader(200)
+		_, err := w.Write(response)
+		if err != nil {
+			return
+		}
+	}))
+
+	return srv, func() {
+		srv.Close()
+	}
+}
+
+func TestGetTokensSuccessRace(t *testing.T) {
+	expected := []GeckoToken{
+		{
+			ID:     "ethereum",
+			Symbol: "eth",
+			Name:   "Ethereum",
+		},
+		{
+			ID:     "status",
+			Symbol: "snt",
+			Name:   "Status",
+		},
+	}
+
+	expectedMap := map[string]GeckoToken{
+		"ETH": {
+			ID:     "ethereum",
+			Symbol: "eth",
+			Name:   "Ethereum",
+		},
+		"SNT": {
+			ID:     "status",
+			Symbol: "snt",
+			Name:   "Status",
+		},
+	}
+	response, _ := json.Marshal(expected)
+
+	var delay time.Duration = 1
```

Comment:
It looks to me that `delaysSecs` is one nanosecond. See here https://pkg.go.dev/time#Duration

If you want one second, you have to fix the `setupTest` as I mentioned above and have this changed to
```go
delaySecs := time.Duration(1)*time.Second
```